tests/fixtures/endpoints_handler.py

```diff
@@ -0,0 +1,206 @@
+        that in the returned parameters"""
+
+        # Get the parameter information from the endpoint function signature
+        signature_parameters = inspect.signature(endpoint_function).parameters
+
+        # type-cast based on type-hinting info
+        typed_parameters: dict[str, Any] = {}
+        for parameter_name, value in string_parameters.items():
+            parameter_type = signature_parameters[parameter_name].annotation
+
+            # all parameters should be typed, raise exception otherwise
+            if parameter_type is inspect.Parameter.empty:
+                raise Exception(
+                    f"Parameter '{parameter_name}' in function "
+                    + f"'{endpoint_function.__name__}' is missing type information!"
+                )  # pylint: disable=broad-exception-raised
+
+            if parameter_type is not str:
+                value = parameter_type(value)
+            typed_parameters[parameter_name] = value
+
+        # include request itself if needed (e.g. for header or auth info),
+        if "request" in signature_parameters:
+            typed_parameters["request"] = request
+
+        logger.info("\tParameters are:")
+        for name, value in typed_parameters.items():
+            logger.info("\t\t%s: %s (%s)", name, value, type(value))
+
+        return typed_parameters
+
+    @classmethod
+    def get_function_and_parameters(
+        cls, url: str, method: str
+    ) -> tuple[Callable, dict[str, str]]:
+        """Iterate through the registered endpoints for the given method.
+        For each registered endpoint, try to match the request's url to the endpoint pattern.
+        Upon matching, return the function and parsed variables from the url (if applicable).
+        """
+        for endpoint in cls.methods[method]:
+            matched_url = re.search(endpoint.url_pattern, url)
+            if matched_url:
+                endpoint_function = endpoint.endpoint_function
+                logger.info("\tGoing to call function: %s", endpoint_function.__name__)
+
+                # return endpoint function with url-string parameters
+                return (
+                    endpoint_function,
+                    matched_url.groupdict(),
+                )
+
+        logger.error("For %s, failed to match URL: `%s`", method, url)
+        assert False
+
+    @classmethod
+    def build_loaded_endpoint_function(cls, request: httpx.Request) -> partial:
+        """Route a request to the correct endpoint, build typed parameter dictionary,
+        and return loaded partial func"""
+
+        # get endpoint function and the parsed string parameters from the url
+        endpoint_function, string_parameters = cls.get_function_and_parameters(
+            url=str(request.url), method=request.method
+        )
+
+        # convert string parameters into the types specified in function signature
+        typed_parameters = EndpointsHandler.convert_parameter_types(
```

Comment:
`cls.convert_parameter_types` would also be possible in Python, though not really sure what best practices has to say with regard to that

Reply:
Changed in [f20e764](https://github.com/ghga-de/ghga-connector/pull/68/commits/f20e76493dd73b2fdb26059258a0bfbb3ded3306)